chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java

```diff
@@ -257,14 +279,17 @@ public Set<byte[]> allKeys() {
   @Override
   public Set<byte[]> allValues() {
     resetDbLock.readLock().lock();
-    try (DBIterator iterator = getDBIterator()) {
-      Set<byte[]> result = Sets.newHashSet();
-      for (iterator.seekToFirst(); iterator.hasNext(); iterator.next()) {
-        result.add(iterator.peekNext().getValue());
+    try {
+      checkDbAlive();
+      try (DBIterator iterator = getDBIterator()) {

```

Comment:
move checkDbAlive into getDBIterator()?

Reply:
check db shall be before any db operations and just need check once actually for one public method ,if put checkdb operation in a private method , the times of checkDb is not sure

Reply:
getDBIterator() is not a db operation?

Reply:
getDBIterator is a db operation and before this method should checkdb but not in this method as we will not use this private method directly